source/virtualBuffers/__init__.py

```diff
@@ -135,11 +135,12 @@ class VirtualBufferTextInfo(browseMode.BrowseModeDocumentTextInfo,textInfos.offs
 	allowMoveToOffsetPastEnd=False #: no need for end insertion point as vbuf is not editable. 
 
 	UNIT_CONTROLFIELD = "controlField"
+	UNIT_FORMATFIELD = "formatField"
 
 	def _getControlFieldAttribs(self,  docHandle, id):
-		info = self.copy()
-		info.expand(textInfos.UNIT_CHARACTER)
-		for field in reversed(info.getTextWithFields()):
+		start, end = self._getOffsetsFromFieldIdentifier(docHandle, id)

```

Comment:
> `VirtualBufferTextInfo._getControlFieldAtribs` now utilizes the new `_getFieldsInRange` method. The new implementation adds an extra rpc call, but no longer creates a copy of the text info. Honestly the current implementation makes much more sense to me and hopefully fixes #7853.

1. Copying a TextInfo is super cheap (it just copies the offsets). An RPC call is not so cheap. I'm worried this could cause significant performance degradation in the Elements List, since we're obtaining TextInfos for all of those elements but then querying their offsets again. At the very least, the performance needs to be tested on big documents before and after this patch.
2. Why do you think this will fix #7853? The TextInfos created for VirtualBufferQuickNavItems use the same RPC call to fetch offsets. It's possible the offsets shift once the Elements List appears (perhaps due to focus listing), but in that case, the TextInfo will also be broken and we need to re-fetch that too. (Otherwise, move and activate will be broken; I think there are other bugs filed about that.)

I understand why you need a way to do this given arbitrary offsets, but I'm not convinced we always want to fetch the offsets if we know the object is definitely within the TextInfo. Perhaps the method could take a `fieldWithinTextInfo=False` argument or similar and not fetch the offsets in that case.

Reply:
Err, I meant focus moving, not focus listing.